### PR TITLE
Replaces vitiligo, revitiligo with two new symptoms: disfiguration and polyvitiligo.

### PR DESCRIPTION
**Pull request by harmonyn**
## About The Pull Request

Replaces vitiligo and revitiligo with two new symptoms:

1. Polyvitiligo: Adds a random crayon powder (a subtype of colorful reagent) to the host causing a color changing effect. 
2. Disfiguration: Makes the host disfigured, causing them to appear as Unknown. (I originally said prosopagnosia but it turns out that trait doesn't affect what you see while mousing over mobs, so it is kind of pointless as a symptom.) 

**NOTE:** I didn't touch the stats, they have the exact same stats as vitiligo and revitiligo. The only difference is I added a prob(50) to one of the visible_messages because it was extremely spammy in testing.

A remake of https://github.com/tgstation/tgstation/pull/45245, substantially different enough that I figured it would be best to just make a new PR instead of dealing with the merge conflict.

## Why It's Good For The Game

- The vitiligo and revitiligo symptoms have no use to legit virologists and very little use to antagonists, their primary purpose is to griff, and not even the kind of griff that results in interesting "minor IC crime."
- Their other primary purpose is to set off a chain reaction of 3edgy racist memes that derail the round into permanent obnoxiousness. 
- These replacement symptoms preserve the (somewhat niche) functionality of making people difficult to tell apart, as well as the (more important) use of buffing viruses.
- The creator himself said the original symptoms were a "joke symptom I threw in as an afterthought": https://github.com/tgstation/tgstation/pull/7197

## Changelog
:cl: bandit
del: Cosmetic mutations have been observed in vitiligo-related viral strains.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
uhh i think you forgot to code is in disfiguration

**Comment by nemvar** [comment]
Nah, he just forgot to commit a file.

File: code/datums/diseases/advance/symptoms/disfiguration.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		else
+			M.visible_message("<span class='warning'>[M]'s face begins to contort...</span>", "<span class='notice'>Your face feels wet and malleable...</span>")
+
+
+/datum/symptom/disfiguration/End(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/carbon/M = A.affected_mob
+	if(M && HAS_TRAIT(M, TRAIT_DISFIGURED))
+		REMOVE_TRAIT(M, TRAIT_DISFIGURED, DISEASE_TRAIT)
```
Comment (nemvar) [inline code/datums/diseases/advance/symptoms/disfiguration.dm]: Traits can be on everything. That's what's so nice about them.
This entire thing should look more like this:
```
/datum/symptom/disfiguration/End(datum/disease/advance/A)
	. = ..()
	if(!.)
		return
	var/mob/living/carbon/M = A.affected_mob
	if(A.affected_mob)
		REMOVE_TRAIT(A.affected_mob, TRAIT_DISFIGURED, DISEASE_TRAIT)

File: code/datums/diseases/advance/symptoms/disfiguration.dm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	resistance = 0
+	stage_speed = 3
+	transmittable = 1
+	level = 5
+	severity = 1
+	symptom_delay_min = 25
+	symptom_delay_max = 75
+
+/datum/symptom/disfiguration/Activate(datum/disease/advance/A)
+	if(!..())
```
Comment (nemvar) [inline code/datums/diseases/advance/symptoms/disfiguration.dm]: Same thing with the . = ..() here.

File: code/datums/diseases/advance/symptoms/skin.dm [hunk 3]
```diff
@@ -72,17 +26,13 @@ BONUS
...
-			return
-		switch(A.stage)
-			if(5)
-				H.skin_tone = "african2"
-				H.update_body(0)
-			else
-				H.visible_message("<span class='warning'>[H] looks a bit dark...</span>", "<span class='notice'>Your skin suddenly appears darker...</span>")
+	switch(A.stage)
+		if(5)
+			M.reagents.add_reagent(/datum/reagent/colorful_reagent, 30)
```
Comment (nemvar) [inline code/datums/diseases/advance/symptoms/skin.dm]: This is actually fairly dangerous. You might overload someone with 1000 units of colorful reagents and then they won't be able to take other chems

Comment (bgobandit) [reply code/datums/diseases/advance/symptoms/skin.dm]: no, should probably cap it at something

**Comment by OverlyGeneric** [comment]
it's meant to be a buffer symptom or harmless fluff for your prank disease.

not all parts of virology have to be debilitating or helpful to the station.

**Comment by PKPenguin321** [comment]
Could use a more descriptive changelog

Also,
>no Simpsonitis

**Comment by Dorsisdwarf** [comment]
Can we get a variant symptom that uses Quantum Hairdye instead?

**Comment by bgobandit** [comment]
fine with it in principle, that said I cannot into viro as much as other people, so would adding a new medium symptom screw with balance

**Comment by Krysonism** [comment]
The colorful reagent overlay is so ugly and obnoxious, these symphony are as griffy as the old ones but lamer.

We should not shit up features to soothe one American liberal's race neurosis.

**Comment by Adanps** [comment]
A polyvitiligo and disfiguration virus could be used to grief, maybe more so than the old symptoms. I do find these to be more interesting than what they would replace. In my ideal world these would be added and the old ones would stay because I think virology needs more options if anything. As for balance, you have to be careful when changing stats because of all the important breakpoints viruses can have, but since the stats are exactly the same it shouldn't matter.

**Comment by bgobandit** [comment]
yeah I'm probably going to change colorful reagent to a random crayonpowder (there isn't currently a random crayonpowder but that is easy enough to add), the effects of those are comparatively muted

**Comment by daklj** [comment]
replaces funny useless symptom with unfunny annoying useless symptom
stupid????

**Comment by bgobandit** [comment]
Have changed the colorful reagent to a random crayon powder, which is a much less visually obnoxious effect

**Comment by PKPenguin321** [comment]
Instead of colorful reagent make it simpsonitis this isn't hard

**Comment by PKPenguin321** [comment]
Do you have me blocked or something ALL I WANT IS TO PLAY AS A SIMPSONS CHARACTER OK

**Comment by bgobandit** [comment]
yellow is one of the crayon reagents

File: code/datums/diseases/advance/symptoms/disfiguration.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/M = A.affected_mob
+	if (HAS_TRAIT(M, TRAIT_DISFIGURED))
+		return
+	switch(A.stage)
+		if(5)
+			ADD_TRAIT(M, TRAIT_DISFIGURED, DISEASE_TRAIT)
+		else
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/diseases/advance/symptoms/disfiguration.dm]: might be better off to remove this so you can get the message when the trait is added.

Comment (bgobandit) [reply code/datums/diseases/advance/symptoms/disfiguration.dm]: added a message here

**Comment by ExcessiveUseOfCobblestone** [comment]
When I crusade against Lexia DarkPurple's tyranny this will have to be addressed as well, but that's a todo.

**Pull request merged**